### PR TITLE
[FLINK-27606][table] Fix code generator exception when using imperative udaf that is introduced for planner scala-free

### PR DESCRIPTION
**Pull request by lsyldliu**
## Brief change log

  - *Fix code generator exception when using imperative udaf that is introduced for planner scala-free*

### CODE REVIEW

**Comment by lsyldliu** [comment]
CC @wuchong

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SQLCodegenE2EITCase.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                            result,
+                            lines.size(),
+                            deadline.timeLeft().getSeconds());
+                }
+            } else {
+                LOG.info("The target Json {} does not exist now", result);
+            }
+            Thread.sleep(500);
+        }
+        Assert.assertTrue("Did not get expected results before timeout.", success);
```
Comment (wuchong) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SQLCodegenE2EITCase.java:183]: Please print the content of the result file if there is any. It would help to debug if the test is not passed on Azure.

**Comment by wuchong** [comment]
Besides, please rebase your branch to the latest master when you addressing the comments.

**Review by wuchong — APPROVED** [review]
LGTM

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                            <!-- override reuseForks to true to reduce testing time -->
+                            <reuseForks>true</reuseForks>
+                            <systemPropertyVariables>
+                                <rootDir>${project.basedir}/../../</rootDir>
+                                <moduleDir>${project.basedir}</moduleDir>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml:119]: Why was this added? Why not just run it in the e2e test module?

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-test-utils-junit</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <scope>compile</scope>
+        </dependency>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml:48]: Why are these `compile` dependencies when this module only contains test classes?

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/PlannerScalaFreeITCase.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+    private static final Path sqlToolBoxJar = TestUtils.getResource(".*SqlToolbox.jar");
+
+    public PlannerScalaFreeITCase(String executionMode) {
+        this.executionMode = executionMode;
+    }
+
+    @Before
+    public void before() throws Exception {
+        DOWNLOAD_CACHE.before();
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/PlannerScalaFreeITCase.java:108]: this is unnecessary

File: flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml (hunk 2 above)
Comment (wuchong) [reply flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml:119]: Yes. We should remove this. This also results in that it runs twice in e2e_2.

Comment (wuchong) [reply flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml:119]: @lsyldliu could you open a pull request to fix this?
